### ISSUE TITLE
yii.js handleAction - add ability to specify submit url for a button without form

### ISSUE THREAD

**Issue by PowerGamer1** [yiisoft/yii2#10672]
The following code renders an anchor ('a' element):
```PHP
Html::a('Click me', ['controller/action', 'someParam' => 'someValue'], 'data-method' => 'post']);
```
For various reasons (for ex., a button can easily be disabled/reenabled by JavaScript code unlike an 'a' element) I need it to be an actual 'button' element AND I do not want to put that button inside a form (for ex., putting a button into a form breaks the CSS styles if several buttons are placed in a row - some inside a form and some not), but if I change it to a button, I lose the ability to specify the url where dynamically generated form will be submitted.

So I propose to implement an ability for yii.js handleAction() function to recognize an url specified through 'data-action', 'data-url' or 'data-href' parameters, like this:

```PHP
Html::button('Click me', ['data-action' => Url::to(['controller/action', 'someParam' => 'someValue']), 'data-method' => 'post']);
```

**Comment by samdark** [comment]
It's out of our scope.

**Comment by PowerGamer1** [comment]
What do you mean by "It's out of our scope." ?

**Comment by samdark** [comment]
I mean that it's quite custom thing that should not be handled by framework.

**Comment by PowerGamer1** [comment]
@samdark Its no more custom that the current functionality of yii.js handleAction, such as 'data-params' or 'data-method' or 'data-confirm' or the fact that yii.js handleAction already can create a form on the fly and submit it. I only propose a way to explicitly specify an url for handleAction. An url that can ALREADY be specified for handleAction but only indirectly (through a form around a button or as href on 'a' element).

**Comment by SilverFire** [comment]
I think @PowerGamer1 describes a valid use case and it might me implemented.

**Comment by samdark** [comment]
Well, at least `data-action` should be renamed to `data-href` or something like that since action is more generic thing.

**Comment by PowerGamer1** [comment]
`handleAction` currently uses `data-method` as in `<form method="">` so `data-action` as in `<form action="">` is a logical choice IMO. And the comment for handleAction already speaks about "form action":

 > For hyperlinks, the **form action** will take the value of the "href" attribute of the link.
For other elements, either the containing **form action** or the current page URL will be used
as the **form action URL**.

But I don't mind against other naming choices for this attribute.

**Comment by samdark** [comment]
Nope. For example, if the button is within the for, it would be pretty confusing. Should it submit the form using action specified or replace location with it?

**Comment by PowerGamer1** [comment]
@samdark
Currently, window.location is replaced only if `data-method` is not specified. Also, if `data-method` is not specified, `data-params` is ignored. The same should go for `data-action` - `data-action` is ignored unless `data-method` is specified. This would be inline with current `handleAction` design and should not confuse anyone.
